devenv.nix

```diff
@@ -0,0 +1,57 @@
+{ pkgs, lib, config, inputs, ... }:
+
+{
+  # https://devenv.sh/packages/
+  packages = [
+    pkgs.git
+    pkgs.gnumake
+  ];
+
+  # https://devenv.sh/languages/
+  languages.python.enable = true;
+  languages.python.poetry.enable = true;
+  languages.python.poetry.activate.enable = true;
+  # languages.python.version = "3.8";
+
+  # https://devenv.sh/scripts/
+  scripts.hello.exec = ''
+    echo "Run 'setup' to prepare this repository for development"
+  '';
+  scripts.setup.exec = ''
+    make dev
+    make help
+  '';
+
+  enterShell = ''
+    hello
+  '';
+
+  # https://devenv.sh/tests/
+  enterTest = ''
+    echo "Running tests"
+    git --version | grep --color=auto "${pkgs.git.version}"
+    make has-poetry
+  '';
+
+  # https://devenv.sh/pre-commit-hooks/
+  git-hooks.hooks.lint = {
```

Comment:
this runs the linter before commit. Can be omitted by passing `-n`, however I am also happy to remove this, if in question.